File: 0253-meeting-rooms-ii/0253-meeting-rooms-ii.java
class Solution {
    public int minMeetingRooms(int[][] intervals) {
        Arrays.sort(intervals, (i1, i2) -> i1[0] - i2[0]);
        if (intervals.length == 1) {
            return 1;
        }

        PriorityQueue<Integer> queue = new PriorityQueue<>();
        queue.add(intervals[0][1]);
        for (int i = 1; i < intervals.length; i++) {
            if (intervals[i][0] >= queue.peek()) {
                queue.remove();
            }

            queue.add(intervals[i][1]);
        }

        return queue.size();
    }
}

File: 0253-meeting-rooms-ii/0253-meeting-rooms-ii.py
class Solution:
    def minMeetingRooms(self, intervals: List[List[int]]) -> int:
        if len(intervals) == 1:
            return 1
        n = len(intervals)
        intervals.sort()

        minHeap = []
        heapq.heappush(minHeap, (intervals[0][1]))
        
        for i in range(1, n):
            if intervals[i][0] >= minHeap[0]:
                heapq.heappop(minHeap)

            heapq.heappush(minHeap, intervals[i][1])

        return len(minHeap)